### PR TITLE
Dont respect cache headers when $this->respectCacheHeaders=false

### PR DESCRIPTION
**Pull request by Nyholm**

### CODE REVIEW

File: src/CachePlugin.php [hunk 1]
```diff
@@ -89,7 +89,9 @@ protected function isCacheable(ResponseInterface $response)
         if (!in_array($response->getStatusCode(), [200, 203, 300, 301, 302, 404, 410])) {
             return false;
         }
-        if ($this->getCacheControlDirective($response, 'no-store') || $this->getCacheControlDirective($response, 'private')) {
+        if ($this->respectCacheHeaders &&
```
Comment (sagikazarmark) [inline src/CachePlugin.php]: I think that this check could be in one return statement.

**Comment by Nyholm** [comment]
I've updated the PR according to your comments

**Comment by sagikazarmark** [comment]
Actually I meant the first condition as well. :smile: 

You could set some helper variables (like the status code list).

**Comment by sagikazarmark** [comment]
`($this->getCacheControlDirective($response, 'no-store') || $this->getCacheControlDirective($response, 'private'))`

This could go into a separate method, like `checkCacheControlCacheble` or `checkCacheControlAllowsCache` or something like that.

**Comment by Nyholm** [comment]
ah. Okey. =)

I'll update again and make sure it is readable.

**Comment by Nyholm** [comment]
hm, okey. I put it in a variable now since it not reused.

File: src/CachePlugin.php [hunk 2]
```diff
@@ -86,14 +86,11 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
...
-        }
-        if ($this->getCacheControlDirective($response, 'no-store') || $this->getCacheControlDirective($response, 'private')) {
-            return false;
-        }
+        $cachableCodes = [200, 203, 300, 301, 302, 404, 410];
+        $privateHeaders = $this->getCacheControlDirective($response, 'no-store') || $this->getCacheControlDirective($response, 'private');
 
-        return true;
+        // If http status code is cachable and if we respect the headers, make sure there is no private cache headers.
+        return in_array($response->getStatusCode(), $cachableCodes) && (!$this->respectCacheHeaders || !$privateHeaders);
```
Comment (sagikazarmark) [inline src/CachePlugin.php]: Hm...

Isn't it the same as `!($this->respectCacheHeaders && $privateHeaders)`

Comment (Nyholm) [inline src/CachePlugin.php]: It is the same according to De Morgan's law. You are correct.

Comment (sagikazarmark) [inline src/CachePlugin.php]: > De Morgan's law

I knew that there is a name for this, but I couldn't recall.

**Comment by sagikazarmark** [comment]
Thanks @Nyholm

**Comment by Nyholm** [comment]
Thank you for merging.

File: src/CachePlugin.php [hunk 3]
```diff
@@ -86,14 +86,11 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
...
-        }
-        if ($this->getCacheControlDirective($response, 'no-store') || $this->getCacheControlDirective($response, 'private')) {
-            return false;
-        }
+        $cachableCodes = [200, 203, 300, 301, 302, 404, 410];
+        $privateHeaders = $this->getCacheControlDirective($response, 'no-store') || $this->getCacheControlDirective($response, 'private');
 
-        return true;
+        // If http status code is cachable and if we respect the headers, make sure there is no private cache headers.
+        return in_array($response->getStatusCode(), $cachableCodes) && !($this->respectCacheHeaders && $privateHeaders);
```
Comment (dbu) [inline src/CachePlugin.php:93]: i think the previous code with early returns was more readable. i will do another small PR :-)

Comment (dbu) [inline src/CachePlugin.php:93]: see #22

**Pull request merged**